build.rs

```diff
@@ -16,19 +46,54 @@ fn main() {
     } else {
         "2.0.0"
     };
-    let lib = pkg_config::Config::new()
-        .atleast_version(required_version)
-        .statik(true)
-        .probe("hwloc")
-        .expect("Could not find a suitable version of hwloc");
-    if cfg!(target_family = "unix") {
-        for link_path in lib.link_paths {
-            println!(
-                "cargo:rustc-link-arg=-Wl,-rpath,{}",
-                link_path
-                    .to_str()
-                    .expect("Link path is not an UTF-8 string")
-            );
+    #[cfg(feature = "bundled")]
+    {
+        let target = env::var("TARGET").expect("Cargo build scripts always have TARGET");
+        let target_os = get_os_from_triple(target.as_str()).unwrap();
+        let hwloc2_source_path = PathBuf::from(env::var("CARGO_MANIFEST_DIR").unwrap());
+        init_submodule(hwloc2_source_path.as_path());
+        let mut hwloc2_compiled_path: PathBuf =
+            compile_hwloc2(hwloc2_source_path.as_path(), target_os);
+        hwloc2_compiled_path.push("lib");
+        println!(
+            "dummy: {}",
+            hwloc2_compiled_path
+                .to_str()
+                .expect("Link path is not an UTF-8 string")
+        );
+        println!(
+            "cargo:rustc-link-arg=-Wl,-rpath,{}",
+            hwloc2_compiled_path
+                .to_str()
+                .expect("Link path is not an UTF-8 string")
+        );
+        println!(
+            "cargo:rustc-link-search={}",
+            hwloc2_compiled_path
+                .to_str()
+                .expect("Link path is not an UTF-8 string")
+        );
+        println!("cargo:rustc-link-lib=static=hwloc");
+        println!("cargo:rustc-link-lib=xml2");
+        println!("cargo:rustc-link-lib=pciaccess");
+        println!("cargo:rustc-link-lib=udev");
```

Comment:
Please also emit one or more [`cargo:rerun-if-changed`](https://doc.rust-lang.org/cargo/reference/build-scripts.html#cargorerun-if-changedpath) directives so the configure/build cycle is not re-run on every `cargo build`. This should include the source dir, and the build dir if you decide to make it separate (see above).

Reply:
I added one directive for rerun-if-changed.